src/org/opendatakit/briefcase/ui/reused/KeyAdapterBuilder.java

```diff
@@ -0,0 +1,49 @@
+package org.opendatakit.briefcase.ui.reused;
+
+import java.awt.event.KeyAdapter;
+import java.awt.event.KeyEvent;
+import java.util.function.Consumer;
+
+public class KeyAdapterBuilder {
+  private Consumer<KeyEvent> keyTypedConsumer;
+  private Consumer<KeyEvent> keyPressedConsumer;
+  private Consumer<KeyEvent> keyReleasedConsumer;
+
+  public KeyAdapterBuilder onKeyTyped(Consumer<KeyEvent> consumer) {
+    keyTypedConsumer = consumer;
+    return this;
+  }
+
+  public KeyAdapterBuilder onKeyPressed(Consumer<KeyEvent> consumer) {
+    keyPressedConsumer = consumer;
+    return this;
+  }
+
+  public KeyAdapterBuilder onKeyReleased(Consumer<KeyEvent> consumer) {
+    keyReleasedConsumer = consumer;
+    return this;
+  }
+
+  public KeyAdapter build() {
+    return new KeyAdapter() {
```

Comment:
Please see my comment about the focus adapter. No need for an adapter here, since you’re implementing everything in the listener interface.

Reply:
I've opted to remove the unused methods instead :P